### PR TITLE
Added PS scripts for enabling to run TS on Windows

### PR DESCRIPTION
**Pull request by olukas**
Hazelcast Go client is currently not supported on Windows. Following powershell scripts provide ability to execute testsuite on Windows which will be useful once Windows will start to be supported. All these scripts are based on similar bash scripts used for testing on linux.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -0,0 +1,47 @@
+$ClientImportPath="github.com/hazelcast/hazelcast-go-client"
+$ListOfPackages = New-Object System.Collections.ArrayList
+go list $ClientImportPath/... | Select-String -pattern ".*/test|.*/compatibility|.*/rc|.*/sample" -notmatch | %{$ListOfPackages.add($_)}
+$PackageList=$ListOfPackages -join ","
+
+#run linter
+cd $env:gopath\src\$ClientImportPath
+.\linter.ps1
+
+# currently comment out due to a lot of found issues
```
Comment (SaitTalhaNisanci) [inline build.ps1]: Why does it find a lot of issues? Can you show a few of the found issues?

File: build.ps1 [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+cd $env:gopath\src\$ClientImportPath
+go build
+
+cd $env:gopath
+.\start-rc.ps1
+
+Start-Sleep -s 11
+
+# Run vet tools (Compiler warning plugin)
+go vet $ClientImportPath > vet.txt
```
Comment (SaitTalhaNisanci) [inline build.ps1]: This is actually not necessary, we already do vet analysis with go metalinter. I will remove this from `build.sh` as well.

File: build.ps1 (hunk 1 above)
Comment (olukas) [reply build.ps1]: I have already written about them to Furkan. You can see those issues here - https://hazelcast-l337.ci.cloudbees.com/job/golang-client-windows-temp/1/consoleFull in time 19:16:15.

File: build.ps1 (hunk 2 above)
Comment (olukas) [reply build.ps1]: Ok, thanks, I will remove it

File: build.ps1 (hunk 1 above)
Comment (SaitTalhaNisanci) [reply build.ps1]: I think I found why it is finding issues. I will `gometalinter.json` for windows.

Comment (olukas) [reply build.ps1]: Great, thanks!

**Comment by yuce** [comment]
Thanks @olukas for this PR. The latest code for the Go client that runs the tests work with Windows Subsystem for Linux (tested only with WSL 1). Is that a reasonable workaround?

**Comment by olukas** [comment]
Hi @yuce , this PR was basically an attempt how to help Go client team with Windows testing 3 years ago. I guess this PR is currently very obsolete and can be closed (or reworked by someone).

Regarding to your question - I think it can be used as workaround but if Windows OS is supported for Go client then you should probably consider to use also real Windows for testing.

**Pull request closed without merging**